tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/polyparser/ArcHybridTransitionSystem.scala

```diff
@@ -396,17 +396,23 @@ case class ArcHybridRequestedArcInterpretation(
       case ArcHybridLeftArc(_) =>
         (StackRef(0)(state).headOption, BufferRef(0)(state).headOption) match {
           case (Some(stackFirst), Some(bufferFirst)) =>
-            val otherToken = arcTokens - stackFirst
+            val otherToken = (arcTokens - stackFirst).head
             arcTokens.contains(stackFirst) &&
-              !arcTokens.contains(bufferFirst) && state.stillActive(otherToken.head)
+              !arcTokens.contains(bufferFirst) &&
+              (state.stack.contains(otherToken) ||
+                otherToken == 0 ||
+                (state.bufferPosition > 0 && state.bufferPosition <= otherToken))
           case _ => false
         }
       case ArcHybridRightArc(_) =>
         (StackRef(0)(state).headOption, StackRef(1)(state).headOption) match {
           case (Some(stackFirst), Some(stackSecond)) =>
-            val otherToken = arcTokens - stackFirst
+            val otherToken = (arcTokens - stackFirst).head
```

Comment:
Without knowing the surrounding code...any chance `.head` here will be called on an empty collection? If so we should likely use `headOption` and handle the `None` case, to avoid an exception.